src/api/app/components/workflow_artifacts_per_step_component.rb

```diff
@@ -0,0 +1,89 @@
+class WorkflowArtifactsPerStepComponent < ApplicationComponent
+  with_collection_parameter :artifacts_per_step
+
+  attr_reader :artifacts_per_step, :step, :artifacts
+
+  def initialize(artifacts_per_step:)
+    super
+    @artifacts_per_step = artifacts_per_step
+    @step = @artifacts_per_step.step
+    @artifacts = @artifacts_per_step.artifacts
+  end
+
+  def call
+    parsed_artifacts = JSON.parse(artifacts).deep_symbolize_keys
+
+    case step
+    when 'Workflow::Step::BranchPackageStep'
+      artifacts_for_branch_or_link_package(parsed_artifacts, 'Branched package from ')
+    when 'Workflow::Step::LinkPackageStep'
+      artifacts_for_branch_or_link_package(parsed_artifacts, 'Linked package from ')
+    when 'Workflow::Step::RebuildPackage'
+      artifacts_for_rebuild_package(parsed_artifacts)
+    when 'Workflow::Step::ConfigureRepositories'
+      artifacts_for_configure_repositories(parsed_artifacts)
+    end
+  rescue JSON::ParserError, ActionController::UrlGenerationError => e
+    Airbrake.notify(e, artifacts_per_step_id: artifacts_per_step.id)
+    tag.li { concat("Could not display artifacts for #{step.split('::').last.titleize}") }
+  end
+
+  private
+
+  def artifacts_for_branch_or_link_package(parsed_artifacts, label)
+    source_path = helpers.package_show_path(project: parsed_artifacts[:source_project], package: parsed_artifacts[:source_package])
+    target_path = helpers.package_show_path(project: parsed_artifacts[:target_project], package: parsed_artifacts[:target_package])
+
+    tag.li do
+      concat(label)
+      concat(link_to("#{parsed_artifacts[:source_project]}/#{parsed_artifacts[:source_package]}", source_path))
+      concat(' to ')
+      concat(link_to("#{parsed_artifacts[:target_project]}/#{parsed_artifacts[:target_package]}", target_path))
+      concat('.')
+    end
+  end
+
+  def artifacts_for_rebuild_package(parsed_artifacts)
+    package_path = helpers.package_show_path(project: parsed_artifacts[:project], package: parsed_artifacts[:package])
+
+    tag.li do
+      concat('Rebuilt package ')
+      concat(link_to("#{parsed_artifacts[:project]}/#{parsed_artifacts[:package]}", package_path))
+      concat('.')
+    end
+  end
+
+  def artifacts_for_configure_repositories(parsed_artifacts)
```

Comment:
The links for a `configure_repositories` step do not point to the right project and repositories pages.

See this screenshot of a workflow run for a pull request event:
![artifacts-ui](https://user-images.githubusercontent.com/1102934/152360124-6de40488-225a-4587-8c0a-c881ed6173d6.png)

The project should be `home:bs-team:openSUSE:open-build-service:PR-12131` instead of `home:bs-team`. `Workflow::ArtifactsCollector` has to be changed for `configure_repositories` to rely on `@step.target_project_name`, so the same as the artifacts for `link_package` and `branch_package` steps:
https://github.com/openSUSE/open-build-service/blob/ca9aaad9b5cab038129f4ade4d5a15efb0df7ab6/src/api/app/services/workflows/artifacts_collector.rb#L17-L19

This can be changed in a follow-up PR as it doesn't change the code here.

Reply:
I've created a card in our backlog for this: https://trello.com/c/SWtE1NiH/2500-use-targetprojectname-for-the-artifacts-of-configurerepositories

Reply:
Lucky that you caught this. Fixed in https://github.com/openSUSE/open-build-service/pull/12166